nixos/tests/web-servers/h2o/tls-recommendations.nix

```diff
@@ -0,0 +1,115 @@
+import ../../make-test-python.nix (
+  { lib, pkgs, ... }:
+
+  let
+    domain = "acme.test";
+    port = 8443;
+
+    hello_txt =
+      name:
+      pkgs.writeTextFile {
+        name = "/hello_${name}.txt";
+        text = "Hello, ${name}!";
+      };
+
+    mkH2OServer =
+      recommendations:
+      { pkgs, lib, ... }:
+      {
+        services.h2o = {
+          enable = true;
+          package = pkgs.h2o.override (
+            lib.optionalAttrs
+              (builtins.elem recommendations [
+                "intermediate"
+                "old"
+              ])
+              {
+                openssl = pkgs.openssl_legacy;
+              }
+          );
+          defaultTLSRecommendations = recommendations;
+          hosts = {
+            "${domain}" = {
+              tls = {
+                inherit port recommendations;
```

Comment:
I'm not sure I see the value in a test where the `defaultTLSRecommendations` are the exact same as this host. What are we testing? Wouldn't an implementation that completely ignores the vhost specific tls recommendations still do the exact same thing with this setup?